distributed/protocol/tests/test_serialize.py

```diff
@@ -92,15 +92,26 @@ def test_serialize_bytestrings():
     "typecode", ["b", "B", "h", "H", "i", "I", "l", "L", "q", "Q", "f", "d"]
 )
 def test_serialize_arrays(typecode):
-    a = array(typecode)
-    a.extend(range(5))
+    a = array(typecode, range(5))
+
+    # handle normal round trip through serialization
     header, frames = serialize(a)
     assert frames[0] == memoryview(a)
     a2 = deserialize(header, frames)
     assert type(a2) == type(a)
     assert a2.typecode == a.typecode
     assert a2 == a
 
+    # split up frames to test joining them back together
+    header, frames = serialize(a)
+    (f,) = frames
+    f = ensure_memoryview(f)
+    frames = [f[:1], f[1:2], f[2:-1], f[-1:]]
+    a3 = deserialize(header, frames)
+    assert type(a3) == type(a)
+    assert a3.typecode == a.typecode
+    assert a3 == a
+
```

Comment:
Would a size 0 array result in 0 frames? Could you add a test for it?

Reply:
Probably not

```python
from array import array

from distributed.protocol import serialize

a = array("I", range(5))

h, f = serialize(a)
print(len(f))
```

That said, sure we can test that case.